### DISCUSSION TITLE
How to trigger function/method on selected row (using rowMarkers="checkbox")

### DISCUSSION THREAD

**Discussion by SyedSibtainRazvi** [glideapps/glide-data-grid#780] [General]
Issue Description:

I am using the rowMarkers feature in GlideApps to allow users to select rows in a grid. However, I am having trouble figuring out how to work with the data of the selected rows programmatically. There seems to be a lack of documentation or guidance on this topic.

Steps to Reproduce:

Enable rowMarkers in a GlideApps grid.
Select multiple rows using checkboxes.
Attempt to perform an action on the selected rows programmatically. Let's say console.log all the rows data that have been selected.
Additional Notes:

I have attempted to use the onGridSelectionChange event handler with the following code:

onGridSelectionChange={(selectedCells) => console.log(selectedCells)}
However, the information provided by this event handler is not sufficient for accessing the data of the selected rows. It returns the following object:

{current: undefined, rows: _CompactSelection, columns: _CompactSelection}

Expected Behavior:

It would be helpful if GlideApps provided documentation or examples on how to access and work with the data of selected rows when rowMarkers="checkbox" is enabled.

Thank you
Reagrds

**Comment by jassmith** [comment]
`selectedCells.rows` can be iterated over. It has full type information available in typescript. CompactSelections are similar to arrays but are designed to be self-compacting to minimize memory overhead at the cost of write speed.

**Reply by jassmith** [reply]
For additional clarity, I think you want to do

```ts
for (const rowIndex of selectedCells.rows) { console.log(row) }
```

**Comment by SyedSibtainRazvi** [comment]
Thanks for the information @jassmith 
A couple of things here.
1. I have kept the `rowMarkers` set to both `checkbox` and `number`. When I add the property `onGridSelectionChange`, I am not able to highlight multiple rows. In rows, the items array just contains the information for the first cell.
I would really appreciate it if you can redirect me to some kind of sandbox. Let me know if you need more information

**Reply by islittlezhou** [reply]
const handleGridSelectionChange = (newSel) => {
    let newRows = CompactSelection.empty();
    if (newSel.current !== undefined) {
      newRows = newRows.add([newSel.current.range.y, newSel.current.range.y + newSel.current.range.height]);
    }
    for (const b of newSel.current?.rangeStack ?? []) {
      newRows = newRows.add([b.y, b.y + b.height]);
    }
    setSelection({
      ...newSel,
      rows: newRows
    });
